src/transformers/models/detr/image_processing_detr_fast.py

```diff
@@ -0,0 +1,1743 @@
+# coding=utf-8
+# Copyright 2024 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Fast Image processor class for DETR."""
+
+import functools
+import io
+import pathlib
+from collections import defaultdict
+from typing import Any, Dict, Iterable, List, Optional, Set, Tuple, Union
+
+import numpy as np
+
+from ...image_processing_utils import BatchFeature, get_size_dict
+from ...image_processing_utils_fast import BaseImageProcessorFast, SizeDict
+from ...image_transforms import (
+    center_to_corners_format,
+    corners_to_center_format,
+    id_to_rgb,
+)
+from ...image_utils import (
+    IMAGENET_DEFAULT_MEAN,
+    IMAGENET_DEFAULT_STD,
+    AnnotationFormat,
+    AnnotationType,
+    ChannelDimension,
+    ImageInput,
+    ImageType,
+    PILImageResampling,
+    get_image_size,
+    get_image_type,
+    infer_channel_dimension_format,
+    make_list_of_images,
+    validate_annotations,
+    validate_kwargs,
+)
+from ...utils import (
+    TensorType,
+    is_torch_available,
+    is_torch_tensor,
+    is_torchvision_available,
+    is_torchvision_v2_available,
+    is_vision_available,
+    logging,
+)
+
+
+if is_torch_available():
+    import torch
+    from torch import nn
+
+if is_vision_available():
+    import PIL
+
+
+if is_torchvision_v2_available():
+    from torchvision.io import read_image
+    from torchvision.transforms.v2 import functional as F
+elif is_torchvision_available():
+    from torchvision.io import read_image
+    from torchvision.transforms import functional as F
+
+
+logger = logging.get_logger(__name__)
+
+SUPPORTED_ANNOTATION_FORMATS = (AnnotationFormat.COCO_DETECTION, AnnotationFormat.COCO_PANOPTIC)
+InterpolationModeConverter = {
+    0: F.InterpolationMode.NEAREST,
+    1: F.InterpolationMode.LANCZOS,
+    2: F.InterpolationMode.BILINEAR,
+    3: F.InterpolationMode.BICUBIC,
+    4: F.InterpolationMode.BOX,
+    5: F.InterpolationMode.HAMMING,
+}
+
+
```

Comment:
Yes indeed. I'm not sure if I should import those functions directly from `image_processing_detr.py` instead of using `# Copied from` at this point, wdyt?

Reply:
We typically don't import from another file, but it'd be nice to do it yes - in general we should be able to import from an `...utils` file (say `processing_utils.py`, `image_processing_utils.py`,   `image_processing_utils_fast.py` ) more generic methods instead of redefining them for each model. cc @ArthurZucker for this pattern type as we discussed it before

Reply:
Yep I think it's good to think about how we can refactor a bit!